app/models/starting_child.rb

```diff
@@ -20,10 +20,19 @@ def benefits_end
     @end_date ? @end_date : tax_years[tax_years.keys.sort.last].last
   end
 
-  # Return the next Monday if start_date is not nil.
+  # Return the date of the Monday in the future that is closest to the start_date.
+  # If the start_date is a Monday, use the start_date.
   def adjusted_start_date
     return nil if start_date.nil?
-    start_date + (start_date.wday < 1 ? 1 : (1 - start_date.wday) + 7)
+
+    if start_date.wday < 1
+      start_date + 1.day
+    elsif start_date.wday > 1
+      number_of_days_til_next_monday = ((1 - start_date.wday) + 7)
+      start_date + number_of_days_til_next_monday
+    else
+      start_date
+    end
```

Comment:
Suggestion: I haven't looked at this in detail, but I think you might be able to use [DateAndTime::Calculations#next_week][1] and [Date#monday?][2] to simplify this. Possibly something like this:  `start_date.monday? ? start_date : start_date.next_week(:monday)`.

[1]: http://api.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-next_week
[2]: http://ruby-doc.org/stdlib-2.3.1/libdoc/date/rdoc/Date.html#method-i-monday-3F